### PR TITLE
Yamato inference tests

### PR DESCRIPTION
**Pull request by chriselion**
### Proposed change(s)
Adds support for running inference on .nn files during yamato tests.
Started adding support for .onnx files too, but hit a snag importing the model. I'd like to merge this and add the onnx support after I get some guidance from Barracuda team.

### Useful links (Github issues, JIRA tickets, ML-Agents forum threads etc.)
https://jira.unity3d.com/browse/MLA-1056

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [x] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

File: com.unity.ml-agents/Runtime/Inference/BarracudaModelParamLoader.cs [hunk 1]
```diff
@@ -149,6 +149,16 @@ public static string[] GetOutputNames(Model model)
                 return failedModelChecks;
             }
 
+            foreach (var constantName in TensorNames.RequiredConstants)
```
Comment (chriselion) [inline com.unity.ml-agents/Runtime/Inference/BarracudaModelParamLoader.cs:152]: This prevents null-reference exceptions when trying to load models without the expected constants in them (which was happening for onnx files before a fix on master). This can be moved to another PR, but it's a nice to have.

File: Project/Assets/ML-Agents/Examples/SharedAssets/Scripts/ModelOverrider.cs [hunk 2]
```diff
@@ -105,6 +108,19 @@ void GetAssetPathFromCommandLine()
                 {
                     m_BehaviorNameOverrideDirectory = args[i + 1].Trim();
                 }
+                else if (args[i] == k_CommandLineModelOverrideExtensionFlag && i < args.Length-1)
```
Comment (chriselion) [inline Project/Assets/ML-Agents/Examples/SharedAssets/Scripts/ModelOverrider.cs:111]: I added support to allow overriding with .onnx files, but later found out that that we need to make some other changes in the import code to support this. Waiting to hear back from the Barracuda team the best way to handle this.

File: Project/Assets/ML-Agents/Examples/SharedAssets/Scripts/ModelOverrider.cs [hunk 3]
```diff
@@ -36,6 +37,8 @@ public class ModelOverrider : MonoBehaviour
 
         string m_BehaviorNameOverrideDirectory;
 
+        string m_OverrideExtension = ".nn";
```
Comment (vincentpierre) [inline Project/Assets/ML-Agents/Examples/SharedAssets/Scripts/ModelOverrider.cs:40]: I think the extension should be `nn` not `.nn`. This should make it simpler to pass the extension as command line argument

Comment (chriselion) [reply Project/Assets/ML-Agents/Examples/SharedAssets/Scripts/ModelOverrider.cs]: Updated.

File: Project/Assets/ML-Agents/Examples/SharedAssets/Scripts/ModelOverrider.cs [hunk 4]
```diff
@@ -105,6 +108,19 @@ void GetAssetPathFromCommandLine()
...
+                    m_OverrideExtension = args[i + 1].Trim();
+                    if (m_OverrideExtension.Equals(".onnx", StringComparison.OrdinalIgnoreCase))
+                    {
+                        // Not supported yet - need to update the model loading code to support
+                        Debug.LogError("ONNX loading not supported yet.");
+                        Application.Quit(1);
+#if UNITY_EDITOR
+                        EditorApplication.isPlaying = false;
+#endif
+                    }
```
[the remark this answers is not in the document]
Comment (chriselion) [reply Project/Assets/ML-Agents/Examples/SharedAssets/Scripts/ModelOverrider.cs:125]: Done
